src/components/Datetime.tsx

```diff
@@ -0,0 +1,190 @@
+import { useState, useRef } from 'react';
+
+const Datetime = () => {
+  const [typeOfState, setTypeOfState] = useState<'success' | 'error'>('success');
+  const [typeOfSize, setTypeOfSize] = useState<'sm' | 'md' | 'lg'>('lg');
+  const [typeOfInput, setTypeOfInput] = useState<'datetime-local' | 'date' | 'time'>('datetime-local');
+  const tdsDatetimeRef = useRef<HTMLTdsDatetimeElement>(null);
+  const [noMinWidth, setNoMinWidth] = useState(false); // State for noMinWidth
+	const [isDatetimeDisabled, setIsDatetimeDisabled] = useState(false);
+
+  const handleNewState = (event: Event) => {
+    const newState: 'error' | 'success' = (event.target as HTMLTdsRadioButtonElement).value as 'error' | 'success';
+    setTypeOfState(newState);
+  };
```

Comment:
```suggestion
  const handleNewState = (event: Event) => {
    const newState = (event.target as HTMLTdsRadioButtonElement).value as 'error' | 'success';
    setTypeOfState(newState);
  };
```

The type on newState is not necessary as you're already casting the value to the same type. (same in other event handlers)